### PR TITLE
Rename rate_limiter_priority to io_priority in IOOptions

### PR DESCRIPTION
**Pull request by gitbw95**
### Summary
In IOOptions, rename rate_limiter_priority to io_priority since rate limiting is one form of QoS, and different file systems may have different ways of interpreting IO priority.
In IOOptions, prio is removed because the prio field is a placeholder right now, we don't populate it, and nobody is looking at it. The corresponding IOPriority (in file_system.h) is removed.

### Test Plan
Unit Tests.

### CODE REVIEW

File: include/rocksdb/file_system.h [hunk 1]
```diff
@@ -86,14 +76,10 @@ struct IOOptions {
   // Timeout for the operation in microseconds
   std::chrono::microseconds timeout;
 
-  // Priority - high or low
-  IOPriority prio;
-
   // Priority used to charge rate limiter configured in file system level (if
```
Comment (hx235) [inline include/rocksdb/file_system.h:79]: @anand1976 - consider we are now calling this "io_priority" (a more general name), should we revise the comment  "Priority used to charge rate limiter configured in file system level" to be more general?

Comment (hx235) [reply include/rocksdb/file_system.h:79]: Thinking of something like:
"The priority of each IO, which will be initialized to the value RocksDB decides for that IO based on various factors when passed to the underlying file system level".

Speaking of "decided ... based on various factors", which is basically the table in PR summary of https://github.com/facebook/rocksdb/pull/9996,https://github.com/facebook/rocksdb/pull/9988, I realized users have no way to know that just by looking at any API **in the code**. 

Is it supposed to be known by users? If it is, should we improve the API doc in the code?

**Review by ajkr — CHANGES_REQUESTED** [review]
I would prefer not landing IOOptions rate_limiter_priority->io_priority name change before making the corresponding change in ReadOptions/WriteOptions. Otherwise the API is inconsistent for the intervening releases.

File: include/rocksdb/file_system.h (hunk 1 above)
Comment (anand1976) [reply include/rocksdb/file_system.h:79]: I agree. Something like "Priority of the IO relative to other RocksDB IO. The file system implementation should consider this when determining the appropriate QoS for the IO request (including, but not limited to, rate limiting)." might be more accurate.

File: include/rocksdb/file_system.h [hunk 2]
```diff
@@ -86,14 +76,10 @@ struct IOOptions {
   // Timeout for the operation in microseconds
   std::chrono::microseconds timeout;
 
-  // Priority - high or low
-  IOPriority prio;
```
Comment (anand1976) [inline include/rocksdb/file_system.h:90]: There are a few references to this and rate_limiter_priority in WS code. In order to ease the transition, I suggest marking these fields as "DEPRECATED". We can completely remove these in the next release (7.4), which will give them enough time change their code.

File: HISTORY.md [hunk 3]
```diff
@@ -22,6 +22,7 @@
 * Added a new immutable db options, enforce_single_del_contracts. If set to false (default is true), compaction will NOT fail due to a single delete followed by a delete for the same key. The purpose of this temporay option is to help existing use cases migrate.
 * Introduce `BlockBasedTableOptions::cache_usage_options` and use that to replace `BlockBasedTableOptions::reserve_table_builder_memory` and  `BlockBasedTableOptions::reserve_table_reader_memory`.
 * Changed `GetUniqueIdFromTableProperties` to return a 128-bit unique identifier, which will be the standard size now. The old functionality (192-bit) is available from `GetExtendedUniqueIdFromTableProperties`. Both functions are no longer "experimental" and are ready for production use.
+* In IOOptions, remove prio and rename rate_limiter_priority to io_priority.
```
Comment (hx235) [inline HISTORY.md:25]: I'd add  `` to class/field name so "In `IOOptions`, remove `prio` and rename `rate_limiter_priority` to `io_priority`"

**Review by hx235 — COMMENTED** [review]
Left some comments

File: include/rocksdb/file_system.h (hunk 1 above)
Comment (hx235) [reply include/rocksdb/file_system.h:79]: > I agree. Something like "Priority of the IO relative to other RocksDB IO. The file system implementation should consider this when determining the appropriate QoS for the IO request (including, but not limited to, rate limiting)." might be more accurate.

Make sense - does the file system implementation need to know how we decide the "Priority of the IO relative to other RocksDB IO" through some kind of API doc?

Comment (gitbw95) [reply include/rocksdb/file_system.h:79]: Thanks! I will update the comments.

File: HISTORY.md (hunk 3 above)
Comment (gitbw95) [reply HISTORY.md:25]: Will update.

**Comment by gitbw95** [comment]
@anand1976  @ajkr  @hx235 ,
As discussed, I will abandon this PR and will defer the name changes to the next major release (8.0).

**Comment by gitbw95** [comment]
> The rate_limiter_priority is in the public API. Won't this change break compilation for any users of this flag?

@mrambacher , we have been discussing the name change (rate_limiter_priority -> io_priority) and will debate further before deciding whether to make the changes in the next major release (8.0).

**Pull request closed without merging**